src/EventSubscriber/DrupalSeamlessCilogonEventSubscriber.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\drupal_seamless_cilogon\EventSubscriber\DrupalSeamlessCilogonEventSubscriber
+ */
+
+namespace Drupal\drupal_seamless_cilogon\EventSubscriber;
+
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+/**
+ * Event Subscriber DrupalSeamlessCilogonEventSubscriber
+ */
+class DrupalSeamlessCilogonEventSubscriber implements EventSubscriberInterface {
+
+  const seamlessCookieName = 'access_ci_sso';
+
+  // TODO implement a constructor to initialize statics?
+
+  /**
+   * Event handler for KernelEvents::REQUEST events, specifically to support seamless login by
+   * checking if a non-authenticated user already has the seamless cilogon flow set.
+   */
+  public function onRequest(RequestEvent $event) {
+    $cookie_name = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_name', self::seamlessCookieName);
+
+    $cookie_exists = isset($_COOKIE[$cookie_name]);
+
+    $seamless_debug = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_debug', true);
+    
+    if ($seamless_debug) {
+      $msg =  __FUNCTION__ . "() - \$_COOKIE[$cookie_name] = " 
+        . ($cookie_exists ? print_r($_COOKIE[$cookie_name], true) : ' <not set>')
+        . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+      \Drupal::messenger()->addStatus($msg);
+      \Drupal::logger('seamless_cilogon')->debug($msg);
+    }
+
+
+    //  If the user is authenticated,no need to redirect to CILogin
+    if (\Drupal::currentUser()->isAuthenticated()) {
+      return;
+    }
+
```

Comment:
hi @a-pasquale -- you wrote: "The redirect is specific for cilogon_auth, but we need to be able to support the oidc oath module that operations is using also. In the longer term, we will try to consolidate on one oath module, but for now we will need to support both I think. There should be a check here for the cilogon_auth module."

Is the check below sufficient for this?  Or should it happen elsewhere / additionally?

Reply:
@a-pasquale - still wondering about this.  the check below supports seamless login flow if `cilogon_auth` module is installed.  But per your comments on the jira ticket, do we also need to allow seamless login if the `openid_connect_accessci_client` module is installed too?

Reply:
We can add that logic later. The method of initiating the redirect will be slightly different also.